src/native/libs/System.IO.Compression.Native/CMakeLists.txt

```diff
@@ -64,6 +74,7 @@ if (CLR_CMAKE_TARGET_UNIX OR CLR_CMAKE_TARGET_BROWSER OR CLR_CMAKE_TARGET_WASI)
         target_link_libraries(System.IO.Compression.Native
```

Comment:
[Can't comment above this line] Should I switch SHARED in line 69 to PRIVATE to fix the error saying that wasm does not support -shared? Note that this is inside the `if(GEN_SHARED_LIB)` branch.

I don't know where else we could be causing that error.

Reply:
You should do `set(GEN_SHARED_LIB OFF)` before `FetchContent_MakeAvailable` and rename all usages of `zlibstatic` to `zlib`. Then we won't even try to build the shared library (which will also help reduce the other failures we were seeing).